### PR TITLE
Update stage docs

### PR DESCRIPTION
**Pull request by elnaske**
## What did you change?

This PR updates the stage-related documentation pages.
The changes are mainly aimed at improving the presentation of information by:
- Updating formatting (adding tables, numbering headings, etc.)
- Rewriting and reordering sections to flow better
- Moving overly detailed/technical information to other pages (e.g. relevant configs)
- Removing redundant information

I also made a few general edits to `train_config.md`, but because this PR is already big enough I didn't touch the other config docs. I might update them in a future PR.

---

## Why did you make this change?

To make the docs more friendly to new users.

---

## Is your PR small enough?

Yes: 10 files, ~250 lines changed (deletions minus insertions)

### CODE REVIEW

File: doc/espnet3/stages/create-dataset.md [hunk 1]
```diff
@@ -1,181 +1,203 @@
...
 create_dataset:
-  func: src.create_dataset.create_dataset
+  recipe_dir: ${recipe_dir}
   dataset_dir: ${dataset_dir}
-  # optional args specific to the recipe
-  archive_path: ${recipe_dir}/../../egs2/mini_an4/asr1/downloads.tar.gz
+
+dataset:
+  _target_: espnet3.components.data.data_organizer.DataOrganizer
+  recipe_dir: ${recipe_dir}
```
Comment (Masao-Someki) [inline doc/espnet3/stages/create-dataset.md]: I think we can remove this. Because it is basically the default value and ported from `TEMPLATE/`

File: doc/espnet3/stages/collect-stats.md [hunk 2]
```diff
@@ -1,140 +1,118 @@
...
-The `collect_stats` stage computes dataset statistics (feature shapes and global
-stats) used by training and normalization. For background, motivation, and
-advanced use cases, see
-[Collect Stats Phase Overview](../core/stats-collection.html).
+`collect_stats` computes shape files and feature statistics used by later
+training steps.
+This serves two broad purposes:
 
-## Quick usage
+1. **Shape information for batching**: Precomputing feature lengths lets the iterator adjust batches based on sequence size, which is one of the main ways ESPnet avoids out-of-memory errors.
```
Comment (Masao-Someki) [inline doc/espnet3/stages/collect-stats.md]: I think it would be good to have a link to the dataloader part to explain how the stats would be used (https://masao-someki.github.io/espnet_draft_home_page/espnet3/core/components/dataloader.html)

File: doc/espnet3/stages/collect-stats.md [hunk 3]
```diff
@@ -1,140 +1,118 @@
...
+  ) -> Dict[str, torch.Tensor]:
+      feats, feats_lengths = self._extract_feats(speech, speech_lengths)
+      return {"feats": feats, "feats_lengths": feats_lengths}
+```
+
+## 4. Configuration
+
+The `collect_stats` stage is configured in the same `training.yaml` used for training.
+
+At minimum, the `stats_dir` key must be set to the directionary where the [output files](#outputs) will be dumped.
```
Comment (Masao-Someki) [inline doc/espnet3/stages/collect-stats.md]: I think we don't need link for output files?

File: doc/espnet3/stages/collect-stats.md [hunk 4]
```diff
@@ -1,140 +1,118 @@
...
-    def forward(self, speech, speech_lengths, text, text_lengths, **kwargs):
-        pass
-
-    def collect_feats(self, speech, speech_lengths, **kwargs):
-        feats = speech
-        # *_lengths are populated by the ESPnet collate function.
-        feats_lengths = speech_lengths
-        return {"feats": feats, "feats_lengths": feats_lengths}
-```
+- [Training config](../config/train_config.md)
```
Comment (Masao-Someki) [inline doc/espnet3/stages/collect-stats.md:118]: For related pages, please refer to other pages and use cards!

File: doc/espnet3/stages/train.md [hunk 5]
```diff
@@ -1,153 +1,51 @@
...
-Typical outputs are written under:
+| Section                    | Description                                 | Details                                             |
+| -------------------------- | ------------------------------------------- | --------------------------------------------------- |
+| `task`                     | task entrypoint for ESPnet2-style models    |
+| `model`                    | model definition and normalization settings |
+| `dataset`                  | `train` and `valid` splits                  | [Dataset](./train/dataset.md)                       |
+| `dataloader`               | collate and iterator settings               | [Dataloader + Collate](./train/dataloader.md)       |
+| `trainer`                  | Lightning trainer configuration             | [Trainer](../core/components/trainer.md)            |
+| `optimizer`, `scheduler`   | single-optimizer training path              | [Optimizer + Scheduler](./train/optim_scheduler.md) |
+| `optimizers`, `schedulers` | named multi-optimizer path                  | [Optimizer + Scheduler](./train/optim_scheduler.md) |
```
Comment (Masao-Someki) [inline doc/espnet3/stages/train.md]: It seems except for trainer we got 404. Could you check the link?

File: doc/espnet3/stages/train.md [hunk 6]
```diff
@@ -1,153 +1,51 @@
...
-    dirpath: ${exp_dir}/checkpoints
-    save_top_k: 3
-    monitor: valid/loss
-    mode: min
-```
+- [Training config](../config/train_config.md)
+- [Training dataset config](./train/dataset.md)
+- [Dataloader](./train/dataloader.md)
+- [Optimizer and scheduler](./train/optim_scheduler.md)
+- [Trainer](../core/components/trainer.md)
```
Comment (Masao-Someki) [inline doc/espnet3/stages/train.md]: please use cards!

File: doc/espnet3/stages/inference.md [hunk 7]
```diff
@@ -1,285 +1,332 @@
 ---
```
Comment (Masao-Someki) [inline doc/espnet3/stages/inference.md:1]: For this file, could you add one example for batch inference?
Basically we can set 

```
runner:
  batch_size: 4
```
in the inference.yaml to pass batched index for the runner.
E.g., when we have 

```
class MyInferenceRunner(InferenceRunner):
    @staticmethod
    def forward(idx, dataset=None, model=None, **kwargs):
        data = dataset[idx]
        model_output = model.decode_stream(data["speech"])
        return {
            "utt_id": data.get("uttid", str(idx)),
            "hyp": model_output["text"],
            "ref": data.get("text", ""),
        }
```

then the `idx` will be the list of index, such as `[0, 1, 2, 3]`

File: doc/espnet3/stages/metrics.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
+---
```
Comment (Masao-Someki) [inline doc/espnet3/stages/metrics.md:1]: Could you change the name to metrics.md

File: doc/espnet3/stages/measure.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+The summary file format is:
+
+```text
+<inference_dir>/
+├── metrics.json
+└── test-clean/
+    ├── ref.scp
+    ├── hyp.scp
+    ├── wer_alignment
+    └── cer_alignment
```
Comment (Masao-Someki) [inline doc/espnet3/stages/measure.md]: For these alignment files, it is generated only for the asr tasks that uses espnet3's wer/cer implementation. I think we can remove these files from this tree.

File: doc/espnet3/stages/measure.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+```python
+{
+    "ref": Path("exp/.../inference/<test_name>/ref.scp"),
+    "hyp": Path("exp/.../inference/<test_name>/hyp.scp"),
+}
+```
+where `<test_name>` is the name of the test partition (e.g., `test-clean`).
+
+> [!IMPORTANT]
+> `measure()` does not preload SCP contents into lists. It resolves file paths and passes them directly to each metric.
```
Comment (Masao-Someki) [inline doc/espnet3/stages/measure.md]: Fix the format to the following:

```
::: important
`measure()` does not preload SCP contents into lists. It resolves file paths and passes them directly to each metric.
:::
```

File: doc/espnet3/stages/publish.md [hunk 11]
```diff
@@ -1,80 +1,156 @@
...
 
-- `pack_model`: build a model package directory (README/meta + artifacts)
-- `upload_model`: upload the packed directory (e.g., to Hugging Face Hub)
+This is a two-step process:
 
-## Quick usage
+| Step           | Description                       | Implementation                                                                        |
+| -------------- | --------------------------------- | ------------------------------------------------------------------------------------- |
+| `pack_model`   | Prepares the model files.         | [`espnet3.utils.publish_utils.pack_model`](../../../espnet3/utils/publish_utils.py)   |
+| `upload_model` | Uploads the model to Hugging Face | [`espnet3.utils.publish_utils.upload_model`](../../../espnet3/utils/publish_utils.py) |
```
Comment (Masao-Someki) [inline doc/espnet3/stages/publish.md]: I think the link is broken.
Do you want the link to the docstrings? Then it should be located at `/guide/espnet3/utils/`.
E.g., if you want the pack_model, then we can access at `/guide/espnet3/utils/pack_model.html`

**Review by Masao-Someki — COMMENTED** [review]
Thank you so much @elnaske , I will merge it after you have fixed the comments!
I realized that for the publication and demo stage, I think we should wait until those PRs are merged and all naming/codings are stablized. For other stages it looks very good!

File: doc/espnet3/stages/collect-stats.md (hunk 2 above)
Comment (elnaske) [reply doc/espnet3/stages/collect-stats.md]: Do you know which markdown file that corresponds to? I can't find the one for the dataloader

File: doc/espnet3/stages/train.md (hunk 5 above)
Comment (elnaske) [reply doc/espnet3/stages/train.md]: It looks like those files were removed during the merge. I've added them back in.

File: doc/espnet3/stages/create-dataset.md [hunk 12]
```diff
@@ -198,6 +196,25 @@ manifest representation.
 
 ## Related pages
 
-- [Dataset references and builders](../core/components/datasets.md)
-- [DataOrganizer](../core/components/data-organizer.md)
 - [Training dataset config](./train/dataset.md)
```
Comment (Masao-Someki) [inline doc/espnet3/stages/create-dataset.md]: Could you remove this also?

File: doc/espnet3/core/components/dataloader.md [hunk 13]
```diff
@@ -0,0 +1,462 @@
+---
```
Comment (Masao-Someki) [inline doc/espnet3/core/components/dataloader.md:1]: I just want to check, those additional files are linked from some page? Sorry I cannot find it..
If not I think we can remove those

**Review by Masao-Someki — COMMENTED** [review]
Thanks! I left 2 comments!

File: doc/espnet3/core/components/dataloader.md (hunk 13 above)
Comment (elnaske) [reply doc/espnet3/core/components/dataloader.md:1]: They're linked to from the table in `train.md` that had the broken links ([here](https://github.com/Masao-Someki/espnet/pull/41#discussion_r3369883194)). They got removed between when I rewrote that page and when I merged the branches, so I added them back in to fix the links. If you think they're unnecessary, I can remove them.

Comment (Masao-Someki) [reply doc/espnet3/core/components/dataloader.md:1]: Ah thanks, understand!
I think then it is moved to `components/`. You can remove it and add link to the corresponding pages!

Comment (elnaske) [reply doc/espnet3/core/components/dataloader.md:1]: I updated the links. I couldn't find any information on dataloaders, so I moved the file over to `components/`, if that's ok
